src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/TracePoint.kt

```diff
@@ -283,12 +283,13 @@ internal class MethodCallTracePoint(
         this.thrownException = exception
     }
 
-    fun initializeParameters(parameters: List<String>) {
-        this.parameters = parameters
-    }
-    
-    fun initializeParameterTypes(parameters: List<String>) {
-        this.parameterTypes = parameters
+    fun initializeParametersRepresentationAndType(
```

Comment:
Let's rename to `initializeParameters` and pass `parameters: List<String>` and `parametersTypes: List<String>`. 
Then, in `ManagedStrategy` class you can declare extension function `MethodCallTracePoint.initializeParameters(parameters: List<Any?>)` that will apply necessary string conversions.